fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java

```diff
@@ -124,12 +124,11 @@ public static ViewHelpers getInstance() {
         final Iterator<Triple> versions = getObjects(graph, subject, predicate);
         final Map<String, Node> map = new TreeMap<>();
         final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-        Triple triple;
-        String date;
         while (versions.hasNext()) {
-            triple = versions.next();
-            date = getVersionDate(graph, triple.getObject());
-            String key = isNullOrEmpty(date) ? format.format(new Date()) : date;
+            final Triple triple = versions.next();
+            String key = Optional.ofNullable(getVersionDate(graph, triple.getObject()))
```

Comment:
Then I leave it to you, but I would find a more explicit formulation with the ternary operator easier to read.